File: C#/125. Valid Palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int l = 0;
        int r = s.Length - 1;
        while(l <= r) {
        	char currL = Char.ToLower(s[l]);
        	char currR = Char.ToLower(s[r]);
        	if (!Char.IsLetterOrDigit(currL)) {
        		l++;
                continue;
        	}
            if(!Char.IsLetterOrDigit(currR)) {
        		r--;
                continue;
        	}
        	if(currL == currR) {
                l++;
                r--;
        	}
            else{
                return false;
        	}
        }
        return true;
    }
}

File: Java/125. Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int l = 0;
        int r = s.length() - 1;
        while(l <= r) {
        	char currL = Character.toLowerCase(s.charAt(l));
        	char currR = Character.toLowerCase(s.charAt(r));
        	if (!Character.isLetterOrDigit(currL)) {
        		l++;
                continue;
        	}
            if(!Character.isLetterOrDigit(currR)) {
        		r--;
                continue;
        	}
        	if(currL == currR) {
                l++;
                r--;
        	}
            else{
                return false;
        	}
        }
        return true;
    }
}
